doc/configuration.rst

```diff
@@ -1,18 +1,36 @@
 Configuration
 =============
 
-All configuration can be done by adding configuration files. They are looked for in:
+All configuration can be done by adding configuration files.
 
- * ``/etc/luigi/client.cfg``
- * ``luigi.cfg`` (or its legacy name ``client.cfg``) in your current working directory
- * ``LUIGI_CONFIG_PATH`` environment variable
+Supported config parsers:
+* ``cfg`` (default)
+* ``toml``
 
-in increasing order of preference. The order only matters in case of key conflicts (see docs for ConfigParser.read_). These files are meant for both the client and ``luigid``. If you decide to specify your own configuration you should make sure that both the client and ``luigid`` load it properly.
+You can choose right parser via ``LUIGI_CONFIG_PARSER`` environment variable. For example, ``LUIGI_CONFIG_PARSER=toml``.
+
+Default (cfg) parser are looked for in:
+
+* ``/etc/luigi/client.cfg`` (deprecated)
+* ``/etc/luigi/luigi.cfg``
+* ``client.cfg`` (deprecated)
+* ``LUIGI_CONFIG_PATH`` environment variable
+
+TOML parser are looked for in:
+
+* ``/etc/luigi/luigi.toml``
+* ``luigi.toml``
+* ``luigi/base.toml``
+* ``luigi/local.toml``
```

Comment:
Because implicit better than explicit. Now we can use explicit inheritance via `luigi.cfg` and `LUIGI_CONFIG_PATH=luigi_local.cfg`. Motivation of inheritance I describe in Pull Request: this is useful for configuring default settings, that can be overridden inlocal config.

Reply:
You mean that it uses local first and then base. Which makes sense so you can have local changes that don't conflict with the base config.

Reply:
I'm removing base and local configs. This need be discussed in future for better realization.